[정해성] sprint3

Sea0Soda opened the pull request:
## 요구사항

### 기본

- [x] Github에 PR(Pull Request)을 만들어서 미션을 제출합니다.
- [x] 피그마 디자인에 맞게 페이지를 만들어 주세요.
- [x] React와 같은 UI 라이브러리를 사용하지 않고 진행합니다.

### 체크리스트 [기본]

#### 공통

- [x] Viewport 너비 기준 반응형 디자인 적용
  - [x] PC: 1200px 이상
  - [x] Tablet: 768px 이상 ~ 1199px 이하
  - [x] Mobile: 375px 이상 ~ 767px 이하
- [x] 375px 미만 사이즈는 고려하지 않음
- https://16sprintmission.netlify.app/
#### 랜딩 페이지

- [x] Tablet: 로고 좌측 24px, 로그인 버튼 우측 24px 여백 유지
- [x] Mobile: 로고 좌측 16px, 로그인 버튼 우측 16px 여백 유지
- [x] 화면이 줄어들면 푸터 영역 간격이 줄어듦

#### 로그인 & 회원가입 페이지 공통

- [x] Tablet: PC 디자인 동일하게 유지
- [x] Mobile: 좌우 여백 16px, 요소 최대 너비 400px 제한

### 체크리스트 [심화]

- [x] 랜딩 페이지에 Open Graph 메타 태그 적용
  - [ ] 제목: “판다 마켓”
  - [ ] 설명: “일상의 모든 물건을 거래해보세요”
  - [ ] 주소/이미지: 자유 설정

---

## 주요 변경사항

- 



---

## 스크린샷

![screencapture-127-0-0-1-5500-login-index-html-2025-04-26-06_30_22](https://github.com/user-attachments/assets/beaf510f-844b-4b4c-ae18-baaa041be589)
![screencapture-127-0-0-1-5500-login-signup-html-2025-04-26-06_27_07](https://github.com/user-attachments/assets/5ed8cac9-d094-457d-8eda-9d0f2076aa1c)
![screencapture-127-0-0-1-5500-2025-04-26-06_26_46](https://github.com/user-attachments/assets/ba9d4619-eef4-40c2-b826-9fe830e6de24)

---

## 멘토에게

-
-
- 셀프 코드 리뷰를 통해 질문 이어가겠습니다.

Sea0Soda committed:
스프린트 미션3 1차 완성본

Sea0Soda committed:
스프린트 미션3 2차 완성본

addiescode-sj commented on index.html:
```diff
@@ -44,11 +47,12 @@ <h1>판다마켓</h1>
           <div class="hot_logo">Hot Item</div>
           <div class="hot_main">
-            인기 상품을<br />
+            인기 상품을<span class="line-break"> <br /> </span>
```
줄바꿈 속성을 제어할때 디자인 의도를 살리기위해 br태그는 그대로 쓰시고,
음절 단위로 강제 줄바꿈되는 속성을 사용해보시는게 유연하게 대처할수있으면서도 방식이 더 깔끔할것같아요:)

https://developer.mozilla.org/en-US/docs/Web/CSS/word-break

addiescode-sj commented on index.html:
```diff
@@ -81,44 +89,78 @@ <h1>판다마켓</h1>
+          src="assets/images/Img_home_bottom.png"
+          alt="팬더둘이 서로 손흔들며 소통하고있는 사진"
+          class="banner5_img"
+        />
```
여기 쓰인 이미지도 srcset과 sizes 사용을 고려한다면 반응형 이미지 리소스 최적화가 이루어질수있겠죠? 첫 화면에 보이는 이미지가 아닌 스크롤을 쭉 내려야 보이는 이미지라면 레이지로딩을 적용할수도있고요 :)

[참고](https://velog.io/@kmh060020/%EC%9D%B4%EB%AF%B8%EC%A7%80-%EC%B5%9C%EC%A0%81%ED%99%94-srcset-picture-)

addiescode-sj commented on login/login.css:
```diff
@@ -209,10 +208,45 @@ html {
+
+/* 모바일 화면 (최대 767px)용 반응형 스타일 */
+
+@media (max-width: 767px) {
```
미디어쿼리 구간을 겹치는 부분없이 잘 사용하셨네요! 👍

addiescode-sj commented on login/login.css:
```diff
@@ -209,10 +208,45 @@ html {
+@media (max-width: 767px) {
+  .login_container,
+  .signup_container {
+    width: 100%;
+    max-width: 400px;
+    padding: 0 16px;
```
한가지 개선할점을 알려드리자면, 미디어쿼리를 사용하실때 컨테이너 요소를 이루는 주요 레이아웃 속성 (여백, 간격, 크기) 등은 변수화해서 관리해보는게 어떨까요? 

예시)
```css
:root {
   --container-max-width: 400px;
}

@media (min-width: 768px) and (max-width: 1199px) {
  :root {
    --container-max-width: 640px;
  }
}

.container {
  width: var(--container-width);
}
```

addiescode-sj replied:
기본 스타일은 제일 작은 화면 (모바일) 기준으로 작성하시고,
넓어지는 브레이크포인트별로 필요한 스타일만 재정의되게 유지한다면 불필요한 스타일 재정의를 최소화할수있답니다 :)

addiescode-sj commented on style.css:
```diff
@@ -369,3 +371,451 @@ span {
+
+/* 테블릿 화면 (최대 1199px)용 반응형 스타일 */
+
+@media (max-width: 1199px) and (min-width: 767px) {
```
여기서는 사용하는 미디어쿼리 조건 평가문이 다르네요! 

`@media (min-width: 768px) and (max-width: 1199px)` 일관성을 위해 하나로 맞춰볼까요?

addiescode-sj commented on style.css:
```diff
@@ -369,3 +371,451 @@ span {
+  }
+}
+
+/* 모바일 화면 (최대 767px)용 반응형 스타일 */
```
앞서 코멘트 드린것처럼, 모바일용 반응형 스타일 분기를 따로 만드시는것보다는, 기본 스타일을 모바일 기준으로 작성해주면 불필요한 스타일 재정의를 최소화할수있어요 :)

addiescode-sj reviewed (approved):
수고하셨습니다!

## 주요 리뷰 포인트
- 미디어 쿼리 사용 관련 피드백
- 반응형 이미지 리소스 최적화

Sea0Soda committed:
style: 피드백 기반 스타일 구조 개선 및 반응형 최적화

- word-break 속성 적용으로 줄바꿈 제어 개선
- banner5 이미지에 lazy loading 적용
- 레이아웃 관련 속성 변수화 (--container-max-width 등)
- 미디어쿼리 조건 표현 방식 통일
